api/src/test/java/com/epam/pipeline/manager/HierarchicalEntityManagerTest.java

```diff
@@ -16,348 +16,315 @@
-                false, AclPermission.READ.getMask());
-
-        Map<AclClass, List<AbstractSecuredEntity>> available = hierarchicalEntityManager.loadAvailable(
-                new AclSid(DefaultRoles.ROLE_USER.getName(), false), null);
-        Assert.assertEquals(2, available.size());
-        Assert.assertEquals((int) available.get(AclClass.TOOL).get(0).getMask(), AclPermission.READ.getMask());
+    public void shouldLoadByRoleSid() {
+        folder.setConfigurations(Collections.singletonList(runConfiguration));
+        doReturn(dockerRegistryList).when(mockRegistryManager).loadAllRegistriesContent();
+        doReturn(folder).when(mockFolderManager).loadTree();
+        mockUserContext();
+        mockSid();
+        initAclEntity(dockerRegistryList);
+        initAclEntity(folder);
+        initAclEntity(registry);
+        initAclEntity(toolGroup);
+        initAclEntity(tool,
+                Collections.singletonList(new UserPermission(DefaultRoles.ROLE_USER.getName(), READ_PERMISSION)));
+        initAclEntity(runConfiguration,
+                Collections.singletonList(new UserPermission(DefaultRoles.ROLE_USER.getName(), READ_PERMISSION)));
+
+        final Map<AclClass, List<AbstractSecuredEntity>> available = hierarchicalEntityManager.loadAvailable(
+                new AclSid(DefaultRoles.ROLE_USER.getName(), true), null);
+
+        assertThat(available.size()).isEqualTo(2);
+        assertThat(available.get(AclClass.TOOL).get(0).getMask()).isEqualTo(READ_PERMISSION);
     }
 
     @Test
-    @WithMockUser(username = USER)
-    public void testLoadingByRoleSidWorksWhenLoadForUser() {
-        DockerRegistry registry = registryManager.create(
-                ObjectCreatorUtils.createDockerRegistryVO(TEST_NAME, USER, USER));
-        ToolGroup toolGroup = toolGroupManager.create(
-                ObjectCreatorUtils.createToolGroup(TEST_NAME, registry.getId()));
-        Tool tool = toolManager.create(
-                ObjectCreatorUtils.createTool(TEST_NAME, toolGroup.getId()), false);
-        grantPermission(tool.getId(), AclClass.TOOL, DefaultRoles.ROLE_USER.getName(),
-                false, AclPermission.READ.getMask());
-
-        Folder folder = createFolder(TEST_NAME, null);
-        RunConfiguration runConfiguration = createRunConfiguration(TEST_NAME, folder.getId());
-        grantPermission(runConfiguration.getId(), AclClass.CONFIGURATION, DefaultRoles.ROLE_USER.getName(),
-                false, AclPermission.READ.getMask());
-
-        Map<AclClass, List<AbstractSecuredEntity>> available = hierarchicalEntityManager
-                .loadAvailable(new AclSid(USER2, true), null);
-        Assert.assertEquals(2, available.size());
-        Assert.assertEquals((int) available.get(AclClass.TOOL).get(0).getMask(), AclPermission.READ.getMask());
+    public void shouldLoadByRoleSidWhenLoadForUser() {
+        folder.setConfigurations(Collections.singletonList(runConfiguration));
```

Comment:
It seems that the new test logic is not repeat the old one: permissions shall be granted to group